### PR TITLE
Bump Netty 4.1.108.Final

### PR DESCRIPTION
**Pull request by Madhukar525722**
# :mag: Description
Upgrade the Netty version to fix the CVEs

## Types of changes :bookmark:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Test Plan 🧪

GHAs

---

# Checklist 📝

- [ ] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

**Be nice. Be informative.**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -179,7 +179,7 @@
         <log4j.version>2.20.0</log4j.version>
         <mysql.jdbc.version>8.0.32</mysql.jdbc.version>
         <mockito.version>4.11.0</mockito.version>
-        <netty.version>4.1.100.Final</netty.version>
+        <netty.version>4.1.108.Final</netty.version>
```
Comment (cxzl25) [inline pom.xml:182]: https://github.com/apache/kyuubi/pull/6520

Comment (pan3793) [reply pom.xml:182]: Seems gRPC is good with this Netty version

**Comment by pan3793** [comment]
Thanks, merged to master/1.9.2

**Comment by pan3793** [comment]
reverted in branch-1.9 because Arrow 12 is incompatible with Netty 4.1.103 and above, see https://github.com/apache/arrow/issues/39265

**Pull request closed without merging**